### ISSUE TITLE
25-DiffBlender: Scalable and Composable Multimodal Text-to-Image Diffusion Models

### ISSUE THREAD

**Issue by mjmaher987** [sharif-ml-lab/visual_compositional_generation_survey#27]
## Task: Read the Paper and Do the following 4 steps
1. Highlight the paper systematically (**green** for important notes - **blue** for very important notes that worth mentioning in the final survey - **red** for the sentences that we don’t understand or need further attention in the future) [In Zotero].
2. Write the Wikipedia summary of the paper in the GitHub issue (_comment in this issue_).
3. Extract the information in the [excel file](https://docs.google.com/spreadsheets/d/15_sh6TqtUvZIJo1-RZxQtsG0Nvg0vPG_IFKCWiGMjdY/edit?usp=sharing).
4. Add the related sentences to the [overleaf file](https://www.overleaf.com/8167246262ccxtrzztjvpz). Note that when you add the sentence, don’t forget to write the reference in the format of ”[ #reference number that is mentioned in the Google Doc - Name of the Author (who added this to the overleaf) ]” at the end of the sentence.

**Comment by AlieNiT** [comment]
# Title: DiffBlender: Scalable and Composable Multimodal Text-to-Image Diffusion Models
<img width="888" alt="Screenshot 2023-12-04 at 5 14 09 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/076dfc8d-ca3b-48ca-8266-209f1bc3b8a3">

## Abstract
* Proposes multimodal text-to-image diffusion model to incorporate diverse modalities like sketch, box, style embeddings
* Achieves this via small hypernetworks trained on top of fixed Latent Diffusion Model parameters
* Facilitates convenient scaling when adding new modalities without altering base generative model
* Outperforms existing baselines on quantitative and qualitative evaluations

## Introduction

* Text-to-image diffusion models rely solely on prompts, limiting fine-grained control
* Additional modalities like sketch, depth, style can enhance generation process
* Existing multimodal models have limitations like supporting only similar modalities
* Proposed DiffBlender handles diverse modalities, scales easily, leverages base model knowledge

## Related Works
<img width="888" alt="Screenshot 2023-12-04 at 5 29 28 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/a4e9f8f9-13ef-44b6-848e-5e04b15545c1">

Diffusion models are highly regarded for generating high-quality images, outperforming adversarial models in terms of fidelity and diversity
While prompt conditioning in T2I diffusion models is sufficient for explaining desired images, expressing fine-grained details via text remains a challenge.
Previous works expanded T2I diffusion models to include additional conditioning modalities like sketches, depth maps, boxes, keypoints, etc.
Some studies require training the entire model from scratch, limiting multimodal training or modality extension.
GLIGEN and T2I-Adapter utilized generative priors of pretrained T2I diffusion models effectively for customized image generation.

## Method

### Preliminaries on Latent Diffusion Models
* Brief discussion about LDMs and their loss function
* Proposal of DIFFBLENDER, an extension of SD-based multimodal networks, leveraging diverse modalities by replacing single captions with multiple modalities to address limitations of unimodal conditioning:

$$C = c_{\text{text}} \rightarrow C = \{ c_{text}, c_1, c_2, \ldots \}$$

### DIFFBLENDER
<img width="888" alt="Screenshot 2023-12-04 at 5 53 04 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/81d849dc-7b5a-4285-bfc5-4ee9213a9d2c">

* Classifies modalities into image-form (sketch), spatial tokens (box, keypoints), and non-spatial (color, style):
  - Image-form conditions directly get added to $z_t$.
  - Spatial tokens get involved by Local Self-Attention, and non-spatial tokens fet involved by Global Self-Attention:
  
<img width="633" alt="Screenshot 2023-12-04 at 5 55 57 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/72e5efab-ce3c-491e-a011-1891988b8c20">
  
<img width="674" alt="Screenshot 2023-12-04 at 5 56 15 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/833c66eb-b43f-47f8-9666-4d3c59cd19f1">

* Uses hypernetworks to process each modality type and compose combinations
* Scalable as only small hypernetworks are trained while base model is fixed

###  Towards extending modalities
<img width="889" alt="Screenshot 2023-12-04 at 6 05 31 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/445f035a-32c7-4581-a218-75c829c9ab00">

* Multimodal diffusion models should ideally expand to additional modalities without altering the base architecture.
* DIFFBLENDER consists of three modules for various modalities and allows easy fusion or concatenation of new modalities.
* The proposed method demonstrates scalability and adaptability by simulating modality extension scenarios, accommodating new modalities with minimal computational cost while retaining previous conditioning information.
* The model's extensibility is showcased through its capacity to effortlessly integrate new modalities in the future by training specific parameters, avoiding complete model retraining and enabling user-driven inclusion of input modalities for enhanced performance.

 ### Mode-specific guidance: enhancing controllability
<img width="609" alt="Screenshot 2023-12-04 at 6 07 52 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/8b977e78-06b1-45f0-85e6-688bc8fa38d8">

* In DIFFBLENDER, the model is conditioned on many modalities, not just text. The above equation is a reasonable extension of classifier-free guidance formula.
* $C$ \ $c_m = $ All conditions except for $m_{th}$ one.

## Experiments
* Training uses COCO and derived modalities like sketch, depth maps, color palettes
* Qualitative results: multimodal generation, reference-guided generation, object reconfiguration
<img width="886" alt="Screenshot 2023-12-04 at 6 19 10 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/c8085a3f-9631-4a81-81d8-d8f0ebb6cba3">

* Quantitative comparison: achieves highest fidelity and conditional scores over unimodal baselines
<img width="886" alt="Screenshot 2023-12-04 at 6 21 19 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/dae10324-f577-4927-9f0a-104f8a648db7">

* Analysis of mode-specific guidance and latent interpolations
<img width="886" alt="Screenshot 2023-12-04 at 6 22 10 PM" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/80975118/85455bd5-06c9-4ea9-87e9-082b44ba3823">

## Conclusion

* DIFFBLENDER is introduced as a novel multimodal T2I diffusion model adept at integrating diverse conditioning modalities effectively.
* The model tailors mechanisms like local or global self-attention modules to capture unique aspects of each input modality, ensuring adaptability.
* Comprehensive evaluations establish DIFFBLENDER as surpassing existing unimodal studies, setting new standards in multimodal generation.
* This work significantly reduces labor-intensive tasks by generating customized images based on detailed and varied conditioning modalities. Future research aims to extend DIFFBLENDER's capabilities to include modalities like audio or video frames, enhancing its control and overall capabilities beyond the image domain.
